src/Chapter3.hs

```diff
@@ -910,6 +1006,22 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = MkGold Int
+
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (MkGold lhs) (MkGold rhs) = MkGold (lhs + rhs)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append lhs rhs = lhs ++ rhs
+
+instance (Append a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append (Just lhs) (Just rhs) = Just (append lhs rhs)
+    append (Just lhs) Nothing    = Just lhs
+    append Nothing (Just rhs)    = Just rhs
+    append Nothing Nothing       = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂